### PR TITLE
Support for inherited packed dependency

### PR DESCRIPTION
**Pull request by Boubalou**
If an `inheritedPackedStjsCompile` dependency configuration is provided, the plugin will look for the inherited `config.properties` file to merge the configuration as well as copying the `*-packed.js` files into the JAR package of your own project for easy deployment.

```
dependencies {
    inheritedPackedStjsCompile "com.mirego.stjs.jreemul.jreemulstjs:1.0.0"
}
```

### CODE REVIEW

File: src/main/groovy/com/github/dzwicker/stjs/gradle/StJsPlugin.groovy [hunk 1]
```diff
@@ -47,8 +70,54 @@ public class StJsPlugin implements Plugin<Project> {
...
+
+            task.getLogger().info("Copying inherited packed files to: " + task.outputs.files.iterator().next().absolutePath)
+            def dest = task.outputs.files.iterator().next()
+            if (!dest.isDirectory()) {
+                throw new IllegalStateException("Destination for packed file should be a directory.")
+            }
+
+            def filteredFileIterator = filteredFiles.iterator()
+            while (filteredFileIterator.hasNext()) {
+                def from = filteredFileIterator.next()
```
Comment (wjobin) [inline src/main/groovy/com/github/dzwicker/stjs/gradle/StJsPlugin.groovy]: Closure groovy (FYI)
`filteredFileIterator.each { from ->`

File: src/main/groovy/com/github/dzwicker/stjs/gradle/StJsPlugin.groovy [hunk 2]
```diff
@@ -47,8 +70,54 @@ public class StJsPlugin implements Plugin<Project> {
...
+
+                task.getLogger().info("Copying inherited packed files from: " + from.absolutePath)
+
+                if (!from.isFile()) {
+                    throw new IllegalStateException("Expected packed file is not a valid file.")
+                }
+
+                def newDest = new File(dest.absolutePath + "/" + from.name)
```
Comment (wjobin) [inline src/main/groovy/com/github/dzwicker/stjs/gradle/StJsPlugin.groovy]: FYI:
`dest.absolutePath + "/" + from.name` -> `"${dest.absolutePath}/${from.name}"`
ou sinon: `dest.absolutePath + '/' + from.name`
Mais bon, le switch de java vers groovy fait déjà en sorte que c'est des groovy string partout

File: src/main/groovy/com/github/dzwicker/stjs/gradle/StJsPlugin.groovy [hunk 3]
```diff
@@ -47,8 +70,54 @@ public class StJsPlugin implements Plugin<Project> {
...
+                newDest << from.bytes
+            }
+        }
+    }
+
+    String getIneritedPackedConfigFile(Task task) {
+        if (task.outputs.files.asFileTree.iterator().hasNext()) {
```
Comment (wjobin) [inline src/main/groovy/com/github/dzwicker/stjs/gradle/StJsPlugin.groovy]: Tu devrais pourvoir faire `!task.outputs.files.empty`

**Comment by wjobin** [comment]
:+1:

**Comment by mbaron** [comment]
:+1:

**Pull request merged**